modules/colors.html

```diff
@@ -0,0 +1,212 @@
+              <td></td>
+            </tr>
+            <tr id="HWB">
+              <th>HWB</th>
+              <td></td>
+            </tr>
+          </tbody>
+        </table>
+
+        <p>
+          <label for="color">Select a color: </label
+          ><input type="color" id="color" value="#bada55" />
+        </p>
+        <p>
+          <label for="opacity">Select an opacity: </label
+          ><input
+            type="range"
+            id="opacity"
+            value="1"
+            min="0"
+            max="1"
+            step="0.1"
+          />
+        </p>
+      </dialog>
+    </div>
+    <script>
+      const colorPicker = document.getElementById("color");
+      const opacityPicker = document.getElementById("opacity");
+
+      document.addEventListener("readystatechange", () => {
+        document.querySelector("dialog").show();
+        createColor();
+      });
+
+      colorPicker.addEventListener("change", () => {
+        createColor();
+      });
+
+      opacityPicker.addEventListener("change", () => {
+        createColor();
+      });
+
+      function createColor() {
+        const currentColor = colorPicker.value;
+        const currentOpacity = opacityPicker.value * 1;
+        setBackgroundColor(currentColor, currentOpacity);
+        createHEX(currentColor, currentOpacity);
+        createRGB(currentColor, currentOpacity);
+      }
+
+      function createHEX(color, opacity) {
+        const cell = document.querySelector("#HEX td");
+        if (opacity === 1) {
+          cell.textContent = color;
+        } else {
+          cell.textContent = hexOpacity(color, opacity);
+        }
+      }
+
+      function createRGB(color, opacity) {
+        const cell = document.querySelector("#RGB td");
+        color = color.substring(1, 7);
+        const hexArray = color.match(/.{1,2}/g);
+        const R = parseInt(hexArray[0], 16);
+        const G = parseInt(hexArray[1], 16);
+        const B = parseInt(hexArray[2], 16);
+
+        if (opacity === 1) {
+          cell.textContent = `rgb(${R} ${G} ${B})`;
+        } else {
+          cell.textContent = `rgb(${R} ${G} ${B} / ${opacity})`;
+        }
+        createHSL(R, G, B, opacity);
+      }
+
+      function createHSL(r, g, b, opacity) {
+        const cell = document.querySelector("#HSL td");
+        (r = r / 255), (g = g / 255), (b = b / 255);
+        const cmin = Math.min(r, g, b);
+        const cmax = Math.max(r, g, b);
+        const delta = cmax - cmin;
+        let h = 0,
+          s = 0,
+          l = 0;
+
+        if (delta === 0) h = 0;
+        else if (cmax === r) h = ((g - b) / delta) % 6;
+        else if (cmax === g) h = (b - r) / delta + 2;
+        else h = (r - g) / delta + 4;
+
+        h = Math.round(h * 60);
+
+        if (h < 0) h += 360;
+
+        l = (cmax + cmin) / 2;
+        s = (delta === 0) ? 0 : delta / (1 - Math.abs(2 * l - 1));
+        s = +(s * 100).toFixed(1);
+        l = +(l * 100).toFixed(1);
```

Comment:
If there is a risk that the result of toFixed() is not a number, we should convert it explicitely:
```suggestion
        s = Number((s * 100).toFixed(1));
        l = Number((l * 100).toFixed(1));
```